yarn.config.cjs

```diff
@@ -714,6 +714,35 @@ function expectControllerDependenciesListedAsPeerDependencies(
   }
 }
 
+/**
+ * Filter out dependency ranges which are not to be considered in `expectConsistentDependenciesAndDevDependencies`.
+ *
+ * @param {string} dependencyIdent - The dependency being filtered for
+ * @param {Map<string, Dependency>} dependenciesByRange - Dependencies by range
+ * @returns {Map<string, Dependency>} The resulting map.
+ */
+function getInconsistentDependenciesAndDevDependencies(
+  dependencyIdent,
+  dependenciesByRange,
+) {
+  const ALLOWED_INCONSISTENT_DEPENDENCIES = Object.entries({
+    // '@metamask/foo': ['^1.0.0'],
+  });
```

Comment:
Should this be near the top of the file?

Reply:
I'm thinking co-locating is clearer but I will move it out of the function body in a follow-up!

Reply:
https://github.com/MetaMask/core/pull/4773/files#diff-594447fef9feecf11f1839e337c464b7b857fdb73fde7e18940eaa9cdf5b819d

Reply:
- #4800